src/LayerPlan.cpp

```diff
@@ -1463,7 +1473,7 @@ void LayerPlan::writeGCode(GCodeExport& gcode)
                 gcode.writeRetraction(retraction_config);
                 if (path.perform_z_hop)
                 {
-                    gcode.writeZhopStart(retraction_config.zHop);
+                    gcode.writeZhopStart(z_hop_height);
```

Comment:
Hi @vgribinchuk , I did some testing, and it seems that (the change from original to) this line causes the 'height when switched' to be applied to every retraction z-hop, instead of just those for the nozzle switch. I don't think that's what you meant to do. cheers, Remco.

Reply:
Hi, @rburema ! Thank you for noticing that. I will have a look and check as soon as I can.

Reply:
`z_hop_height = retraction_config.zHop;`
got back this line, since it is inside the `for` loop, and `writeZhopStart` is called multiple times. To not apply the 'height when switched' to every retraction, `writeZhopStart` is called now with correct z-hop value.

Reply:
My original point was that the line _after_ this one doesn't do anything, since `z_hop_height` isn't referred to after that point, and will just reset to the value given by the declaration of `z_hop_height` in the next iteration of the loop. Which had me worried you might've forgotten to commit something. That didn't/doesn't seem to be the case, so I think we can move ahead.

Reply:
Ah I see, there was an inner for-loop after all. Not sure how I could have missed that. Sorry!